codegen/src/main/java/software/amazon/awssdk/codegen/poet/rules/EndpointResolverInterceptorSpec.java

```diff
@@ -295,6 +306,33 @@ private MethodSpec ruleParams() {
         return b.build();
     }
 
+    private MethodSpec getCredentialScopeIfPresent() {
+        MethodSpec.Builder b = MethodSpec.methodBuilder("getCredentialScopeIfPresent")
+                                         .addModifiers(Modifier.PRIVATE, Modifier.STATIC)
+                                         .addParameter(ExecutionAttributes.class, "executionAttributes")
+                                         .returns(Region.class);
+
+        b.addStatement("$T<?> authScheme = executionAttributes.getAttribute($T.SELECTED_AUTH_SCHEME)",
+                       SelectedAuthScheme.class, SdkInternalExecutionAttribute.class);
+        b.beginControlFlow("try");
+
+        b.addStatement("$T identity = $T.joinLikeSync(authScheme.identity())", Identity.class, CompletableFutureUtils.class);
+        b.beginControlFlow("if (identity instanceof $T)", AwsCredentialsIdentity.class);
+        b.addStatement("$T awsCredentialsIdentity = ($T) identity", AwsCredentialsIdentity.class, AwsCredentialsIdentity.class);
+        b.addStatement("return awsCredentialsIdentity.credentialScope().map($T::of).orElse(null)", Region.class);
+        b.endControlFlow();
+
+        b.endControlFlow();
+
+        b.beginControlFlow("catch ($T error)", Exception.class);
```

Comment:
I dont think we need try catch here.

The [joinLikeSync](https://github.com/aws/aws-sdk-java-v2/blob/becd54c88a5ac611957bad242d2823971ac5bd63/utils/src/main/java/software/amazon/awssdk/utils/CompletableFutureUtils.java#L247-L259) already throws a Runtime exception